### PR TITLE
mrc-4177 allow passing of dependencies to task queue

### PR DESCRIPTION
**Pull request by hillalex**
Allows single or bulk enqueuing to accept a vector or list of task ids to depend on. Requires https://github.com/mrc-ide/context/pull/21

### CODE REVIEW

File: R/queue_base.R [hunk 1]
```diff
@@ -250,7 +250,9 @@ queue_base <- R6::R6Class(
     ##'
     ##' @param name Optional name for a created bundle
     ##'
-    ##' @param depends_on Optional vector of task ids to depend on
+    ##' @param depends_on Optional task ids to depend on.
```
Comment (richfitz) [inline R/queue_base.R:253]: I believe that roxygen2 can do `@inheritParams` from a different package so you can do `@inheritParams context::bulk_task_save` here and delete this specific docs and then you'd not need to duplicate the text? Not 100%, give it a try.

Comment (hillalex) [reply R/queue_base.R]: Looks like that doesn't work inside R6 classes which support a more [limited set of tags](https://roxygen2.r-lib.org/articles/rd-other.html#r6). I've updated to link to the `context` docs though.

File: DESCRIPTION [hunk 2]
```diff
@@ -23,7 +23,7 @@ Suggests:
     testthat,
     thor
 Remotes:
-    mrc-ide/context
+    mrc-ide/context@mrc-4177
```
Comment (richfitz) [inline DESCRIPTION:26]:
```suggestion
    mrc-ide/context
```

this can be reset now

**Review by richfitz — APPROVED** [review]
Looks good, just the branch pointer to revert

**Pull request merged**